### PR TITLE
Add external credential support back

### PR DESCRIPTION
**Pull request by hatched**
In rewriting the login mechanism the external auth setup was never re-implemented and this broke HJC.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5999/
Test PASSed.

**Comment by bac** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6005/
Test PASSed.

File: jujugui/static/gui/src/app/store/env/base.js [hunk 1]
```diff
@@ -469,14 +469,14 @@ YUI.add('juju-env-base', function(Y) {
...
-     * @param {Object} The credentials to store, with a "user" and a "password"
-     *   attribute included, or with a "macaroons" attribute, depending on the
-     *   method used to log in. The user must be a user name, not a user tag.
-     * @return {undefined} Stores data only.
-     */
+      Store the user's credentials in session storage.
+
+      @method setCredentials
+      @param {Object} The credentials to store.
+        Possible properties { user, password, macaroons, external }.
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/base.js]: Please mention that user and password must be strings, macaroons is a list of objects and external is an object.

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -2649,7 +2655,7 @@ YUI.add('juju-gui', function(Y) {
        @type {Boolean}
        @default false
        */
-      interactiveLogin: {value: false},
+      interactiveLogin: { value: true },
```
Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: Thanks! Well, except for the space inside braces. But that's how life goes, you'll keep adding them, I'll keep removing them...

File: jujugui/static/gui/src/app/store/env/base.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/store/env/base.js]: Done!

**Comment by hatched** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/6010/
Test PASSed.

**Pull request merged**